server/src/main/java/io/druid/server/initialization/EmitterModule.java

```diff
@@ -80,7 +81,18 @@ public void configure(Binder binder)
   public ServiceEmitter getServiceEmitter(@Self Supplier<DruidNode> configSupplier, Emitter emitter)
   {
     final DruidNode config = configSupplier.get();
-    final ServiceEmitter retVal = new ServiceEmitter(config.getServiceName(), config.getHostAndPort(), emitter);
+    String version = getClass().getPackage().getImplementationVersion();
+    final ImmutableMap<String, String> otherServiceDimensions = ImmutableMap.of(
+        "version",
+        // Version is null during `mvn test`.
+        version != null ? version : "testing-version"
```

Comment:
Since folks in the wild can package or embed this code in strange ways, can this be returned to the prior method of `Strings.nullToEmpty`?